File: 12914.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<long long> memo(2001, 0);

long long calcRecursively(int n) {
    if(memo[n] != 0) return memo[n];
    else return memo[n] = (calcRecursively(n - 1) + calcRecursively(n - 2)) % 1234567;
}

long long solution(int n) {
    for(int i = 1; i < 3; i++) memo[i] = i;   
    
    return calcRecursively(n);
}

File: 12914.swift
func solution(_ n:Int) -> Int {
    var dp: [Int] = Array(repeating: 0, count: 2001)
    dp[1] = 1
    dp[2] = 2
    
    for i in 3...2000 { dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567 }
    
    return dp[n]
}
